src/main/java/io/dropwizard/sharding/dao/LookupDao.java

```diff
@@ -430,6 +428,39 @@ public LockedContext(int shardId, SessionFactory sessionFactory, Function<T, T>
             });
         }
 
+        public<U> LockedContext<T> selectAndUpdateOrSave(RelationalDao<U> relationalDao, DetachedCriteria criteria, Function<U, U> handler) {
+            return apply(parent-> {
```

Comment:
Doesn't seem correct. Always introspect when you are putting `Or` etc in the name. Functions should do one and one thing only. More details below.